File: problems/two-sum/variants/cpp/easy.cpp
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int index = 0; index < static_cast<int>(nums.size()); ++index) {
            // Check for target - nums[index], then remember nums[index].
        }
        return {};
    }
};

File: problems/two-sum/variants/python/easy.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for index, value in enumerate(nums):
            complement = target - value
            # Look up the complement and return both indices when found.
            seen[value] = index
        return []
